cupyx/time.py

```diff
@@ -64,36 +72,66 @@ def repeat(
         raise ValueError('`str` should be a string.')
     if not isinstance(n_warmup, int):
         raise ValueError('`n_warmup` should be an integer.')
+    if not isinstance(devices, tuple):
+        raise ValueError('`devices` should be of tuple type')
+
+    return _repeat(
+        func, args, kwargs, n_repeat, name, n_warmup, max_duration, devices)
+
+
+def _repeat(
+        func, args, kwargs, n_repeat, name, n_warmup, max_duration, devices):
+
+    events_1 = []
+    events_2 = []
+
+    for i in devices:
+        with cupy.cuda.Device(i):
+            events_1.append(cupy.cuda.stream.Event())
+            events_2.append(cupy.cuda.stream.Event())
 
     ev1 = cupy.cuda.stream.Event()
     ev2 = cupy.cuda.stream.Event()
 
     for i in range(n_warmup):
         func(*args, **kwargs)
 
-    ev1.record()
-    ev1.synchronize()
+    for event, device in zip(events_1, devices):
+        with cupy.cuda.Device(device):
+            event.record()
+        event.synchronize()
 
     cpu_times = []
-    gpu_times = []
+    gpu_times = [[] for i in range(len(events_1))]
     duration = 0
     for i in range(n_repeat):
-        ev1.record()
+        for event, device in zip(events_1, devices):
+            with cupy.cuda.Device(device):
+                event.record()
+
         t1 = time.perf_counter()
 
         func(*args, **kwargs)
 
         t2 = time.perf_counter()
-        ev2.record()
-        ev2.synchronize()
         cpu_time = t2 - t1
-        gpu_time = cupy.cuda.get_elapsed_time(ev1, ev2) * 1e-3
         cpu_times.append(cpu_time)
-        gpu_times.append(gpu_time)
+
+        for event, device in zip(events_2, devices):
+            with cupy.cuda.Device(device):
+                event.record()
+        for event, device in zip(events_2, devices):
+            with cupy.cuda.Device(device):
+                event.synchronize()
+        i = 0
+        for ev1, ev2 in zip(events_1, events_2):
+            gpu_time = cupy.cuda.get_elapsed_time(ev1, ev2) * 1e-3
+            gpu_times[i].append(gpu_time)
+            i += 1
```

Comment:
```suggestion
        for i, (ev1, ev2) in enumerate(zip(events_1, events_2)):
            gpu_time = cupy.cuda.get_elapsed_time(ev1, ev2) * 1e-3
            gpu_times[i].append(gpu_time)
```

Reply:
I didn't know I could expand the tuple like this 😬
awesome!